src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -836,14 +838,15 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
    * if one uses `check-passwd: { "type": "previous" }`. There are a few other optimizations
    * too, e.g. using the previous SELinux policy in unified core. Also, we might need the
    * commit *object* for next version incrementing. */
-  if (self->ref)
+  const char *previous_ref = self->ref ?: opt_previous_commit;
```

Comment:
Hmm, I think `--previous-commit` should take precedence over the ref if it's provided.